iceoryx_utils/test/moduletests/test_cxx_vector.cpp

```diff
@@ -767,11 +898,37 @@ TEST_F(vector_test, EraseOfLastElementCallsDTorOnly)
     sut1.emplace_back(7);
     sut1.emplace_back(8);
     sut1.emplace_back(9);
-
+    classValue = 0; // reset, since emplace_back set it
     sut1.erase(sut1.begin() + 2);
 
     EXPECT_THAT(dTor, Eq(1));
-    EXPECT_THAT(classValue, Eq(9));
+    EXPECT_THAT(classValue, Eq(0));
+}
+
+TEST_F(vector_test, EraseOfMiddleElementCallsCorrectDTors)
+{
+    vector<CTorTest, 5> sut1;
+    sut1.emplace_back(7);
+    sut1.emplace_back(8);
+    sut1.emplace_back(9);
+
+    sut1.erase(sut1.begin() + 1);
+
+    EXPECT_THAT(dTor, Eq(2));
+    EXPECT_THAT(dTorClassValue, Eq(9));
+}
+
+TEST_F(vector_test, EraseByIndexOfLastElementCallsDTorOnly)
+{
+    vector<CTorTest, 5> sut1;
+    sut1.emplace_back(7);
+    sut1.emplace_back(8);
+    sut1.emplace_back(9);
+
+    sut1.erase((uint64_t)2);
```

Comment:
Please never use c-style casts. Here you should write `sut1.erase(2U);`

Reply:
Fine. I consequently also changed `erase((uint64_t)0U)` to `erase(static_cast<uint64_t>(0U))`.